src/smoothcon/smoothcon.py

```diff
@@ -1,43 +1,89 @@
 from __future__ import annotations
 
+import typing
+import uuid
+
 import numpy as np
 import pandas as pd
-import rpy2.robjects as ro
-import rpy2.robjects.packages as rpackages
+import polars as pl
 from numpy.typing import ArrayLike
-from rpy2.rinterface import NULL
-from rpy2.robjects import conversion, default_converter, pandas2ri
-
-mgcv = rpackages.importr("mgcv")
+from ryp import options, r, to_py, to_r
+
+# Configure ryp to use polars format (currently the default)
+options(to_py_format="polars")
+
+# Load mgcv package (suppress startup messages)
+r("suppressPackageStartupMessages(library(mgcv))")
+
+
+def _convert_to_polars(
+    data: pd.DataFrame | pl.DataFrame | dict[str, ArrayLike],
+) -> pl.DataFrame:
+    """Convert input data to a polars DataFrame."""
+    match data:
+        case dict():
+            # convert JAX arrays to numpy arrays for polars compatibility
+            converted_data: dict[str, ArrayLike] = {}
+            for key, value in data.items():
+                # check if it's a JAX array
+                if (
+                    hasattr(value, "__module__")
+                    and value.__module__ is not None
+                    and "jax" in value.__module__
+                ):
+                    converted_data[key] = np.asarray(value)
+                else:
+                    converted_data[key] = value
+            return pl.DataFrame(converted_data)
+        case pd.DataFrame():
+            return pl.from_pandas(data)
+        case pl.DataFrame():
+            return data
+        case _:
+            typing.assert_never(data)
 
 
 class SmoothCon:
     def __init__(
         self,
-        spec: str | ro.vectors.ListVector,
-        data: ro.vectors.DataFrame | pd.DataFrame | dict[str, ArrayLike],
-        knots: ArrayLike | ro.FloatVector | None = None,
+        spec: str,
+        data: pd.DataFrame | pl.DataFrame | dict[str, ArrayLike],
+        knots: ArrayLike | None = None,
         absorb_cons: bool = True,
         diagonal_penalty: bool = True,
         scale_penalty: bool = True,
         pass_to_r: dict | None = None,
     ) -> None:
         self.pass_to_r = pass_to_r if pass_to_r is not None else {}
         self.spec = spec
-        self.data_r = data
-        self.knots_r = knots
+        self.data = _convert_to_polars(data)
+        self.knots = knots
         self.absorb_cons = absorb_cons
         self.diagonal_penalty = diagonal_penalty
         self.scale_penalty = scale_penalty
 
-        self.smooth = mgcv.smoothCon(
-            self.spec,
-            data=self.data_r,
-            knots=self._knots_r,
-            absorb_cons=absorb_cons,
-            diagonal_penalty=diagonal_penalty,
-            scale_penalty=scale_penalty,
+        # generate unique variable names for R environment
+        self._data_var = f"smoothcon_data_{uuid.uuid4().hex[:8]}"
```

Comment:
I find the naming with the suffix `_var` to be a mit confusing. Maybe instead `_r_name` ?